libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -4819,20 +4814,42 @@ MAV_RESULT GCS_MAVLINK::handle_command_long_packet(const mavlink_command_long_t
         result = handle_flight_termination(packet);
         break;
 
-    case MAV_CMD_COMPONENT_ARM_DISARM:
-        result = handle_command_component_arm_disarm(packet);
-        break;
-
-    case MAV_CMD_FIXED_MAG_CAL_YAW:
-        result = handle_fixed_mag_cal_yaw(packet);
-        break;
-
 #if AP_LANDINGGEAR_ENABLED
         case MAV_CMD_AIRFRAME_CONFIGURATION:
             result = handle_command_airframe_configuration(packet);
             break;
 #endif
 
+    case MAV_CMD_DO_SET_ROI_LOCATION:
+    case MAV_CMD_DO_SET_ROI:
+    case MAV_CMD_FIXED_MAG_CAL_YAW:
+    case MAV_CMD_COMPONENT_ARM_DISARM: {
+        // try the same thing as a COMMAND_INT:
+        MAV_FRAME frame = MAV_FRAME_GLOBAL_RELATIVE_ALT;
+        if (command_long_stores_location((MAV_CMD)packet.command)) {
+            switch (packet.command) {
+            case MAV_CMD_DO_SET_HOME:
```

Comment:
missing from above list?

Reply:
Vestige of further work which I removed.